File: articles/perseverance-and-passion-for-long-term-goals/go/perseverance_passion_score.go
package main

import "fmt"

func GritScore(perseveranceEffort float64, durablePassion float64) float64 {
	return 0.60*perseveranceEffort + 0.40*durablePassion
}

func main() {
	fmt.Printf("Synthetic perseverance-passion grit score: %.3f\n", GritScore(0.80, 0.65))
}

File: articles/perseverance-and-passion-for-long-term-goals/rust/perseverance_passion_score.rs
fn grit_score(perseverance_effort: f64, durable_passion: f64) -> f64 {
    0.60 * perseverance_effort + 0.40 * durable_passion
}

fn main() {
    println!(
        "Synthetic perseverance-passion grit score: {:.3}",
        grit_score(0.80, 0.65)
    );
}
